shared/metrics/src/main/java/io/pravega/shared/metrics/StatsProviderImpl.java

```diff
@@ -75,9 +77,12 @@ public void start() {
     @Synchronized
     @Override
     public void startWithoutExporting() {
-        for (MeterRegistry registry : metrics.getRegistries()) {
-            metrics.remove(registry);
+
+        Iterator<MeterRegistry> iterator = metrics.getRegistries().iterator();
```

Comment:
I don't understand how this is any different from the code before. `for(X : Y)` is invoking `Y.iterator()` and looping through it.

What am I missing?

Reply:
FWIW, I get the same failure with the new code (my line numbers will be different):
```io.pravega.shared.metrics.MetricsProviderTest > testGauge FAILED
    java.util.ConcurrentModificationException
        at java.util.IdentityHashMap$IdentityHashMapIterator.nextIndex(IdentityHashMap.java:735)
        at java.util.IdentityHashMap$KeyIterator.next(IdentityHashMap.java:826)
        at java.util.Collections$UnmodifiableCollection$1.next(Collections.java:1044)
        at io.pravega.shared.metrics.StatsProviderImpl.startWithoutExporting(StatsProviderImpl.java:85)
        at io.pravega.shared.metrics.StatsProviderProxy.startWithoutExporting(StatsProviderProxy.java:45)
        at io.pravega.shared.metrics.MetricsProviderTest.setUp(MetricsProviderTest.java:35)
```
Is this test working on some machines?

Reply:
the latest change fixes it for me: `io.pravega.shared.metrics.MetricsProviderTest > testGauge PASSED`